src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1132,13 +1132,14 @@ internal virtual void IntersectWithEnumerable(IEnumerable<T> other)
                 if (Contains(item))
                 {
                     toSave.Add(item);
+                    DoRemove(item);
                 }
             }
 
-            if (toSave.Count < Count)
+            Clear();
+            foreach (T item in toSave)
             {
-                Clear();
-                AddAllElements(toSave);
+                AddIfNotPresent(item);
```

Comment:
> there are 0 tests for UnionWith

What do you mean? There are tests for it here:
https://github.com/dotnet/corefx/blob/master/src/Common/tests/System/Collections/ISet.Generic.Tests.cs
https://github.com/dotnet/corefx/blob/master/src/System.Collections/tests/Generic/SortedSet/SortedSet.Generic.Tests.cs#L14

and while code coverage isn't the end-all-be-all of test measurement, UnionWith has 100% line and branch coverage from these tests:
https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/lastSuccessfulBuild/artifact/bin/tests/coverage/System.Collections_SortedSet_1.htm